examples/all-devices-app/esp32/main/CMakeLists.txt

```diff
@@ -0,0 +1,124 @@
+#
+#    Copyright (c) 2025 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+get_filename_component(CHIP_ROOT ${CMAKE_SOURCE_DIR}/third_party/connectedhomeip REALPATH)
+get_filename_component(ALL_DEVICES_COMMON_DIR ${CHIP_ROOT}/examples/all-devices-app/all-devices-common REALPATH)
+
+set(PRIV_INCLUDE_DIRS_LIST
+                      "${ALL_DEVICES_COMMON_DIR}"
+                      "${CHIP_ROOT}/examples/providers"
+)
+
+# Application sources
+set(APP_SRCS
+    "${CMAKE_CURRENT_LIST_DIR}/main.cpp"
+    # All-devices common sources
+    "${ALL_DEVICES_COMMON_DIR}/devices/interface/DeviceInterface.cpp"
```

Comment:
Is this typical of cmake projects or IDF projects?

I wonder if for this app we can make this more modular somehow and start having some `*.cmake`  or individual `CmakeLists.txt` in various folders?